interlok-azure-onedrive/src/main/java/com/adaptris/interlok/azure/onedrive/DocumentDownloadService.java

```diff
@@ -0,0 +1,130 @@
+package com.adaptris.interlok.azure.onedrive;
+
+import com.adaptris.annotation.AdapterComponent;
+import com.adaptris.annotation.ComponentProfile;
+import com.adaptris.annotation.DisplayOrder;
+import com.adaptris.annotation.InputFieldHint;
+import com.adaptris.core.AdaptrisConnection;
+import com.adaptris.core.AdaptrisMessage;
+import com.adaptris.core.ServiceException;
+import com.adaptris.core.ServiceImp;
+import com.adaptris.interlok.azure.GraphAPIConnection;
+import com.microsoft.graph.models.extensions.Drive;
+import com.microsoft.graph.models.extensions.DriveItem;
+import com.microsoft.graph.models.extensions.IGraphServiceClient;
+import com.microsoft.graph.options.Option;
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.io.IOUtils;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+/**
+ * Retrieve the contents of a file from OneDrive.
+ *
+ * @config azure-one-drive-document-download-service
+ */
+@XStreamAlias("azure-one-drive-document-download-service")
+@AdapterComponent
+@ComponentProfile(summary = "Retrieve the contents of a file from OneDrive.", tag = "file,o365,microsoft,office,365,one drive,download")
+@DisplayOrder(order = { "connection", "username", "filename" })
+public class DocumentDownloadService extends ServiceImp
+{
+  /**
+   * Connection to Azure OneDrive.
+   */
+  @Getter
+  @Setter
+  @NotNull
+  private AdaptrisConnection connection;
```

Comment:
@albinoloverats  
- the service needs to manage the lifecycle for the connection, since the connection is a component that has a lifecycle.
- the service should probably implement ConnectedService to help the UI.


```

public void prepare() throws CoreException {
   super.prepare();
   LifecycleHelper.prepare(getConnection():
}
protected void initService() throws CoreException {
   LifecycleHelper.init(getConnection():
}
public void start() throws CoreException {
   super.start();
   LifecycleHelper.start(getConnection():
}
public void stop() {
   super.stp[();
   LifecycleHelper.start(getConnection():
}
protected void closeService() {
   LifecycleHelper.close(getConnection():
}
```

That obviously holds true for all services that wrap a connection.

Reply:
I've now implemented these methods, in both the upload & download services.